src/app/mod.rs

```diff
@@ -52,42 +52,50 @@ type Shared<T> = Arc<RwLock<T>>;
 #[derive(Clone)]
 pub(crate) struct BaseCoinApp<S> {
     store: MainStore<S>,
-    pub modules: Shared<Vec<Box<dyn Module<Store = ModuleStore<S>> + Send + Sync>>>,
-    account: Shared<BaseAccount>, // TODO(hu55a1n1): get from user and move to provable store
+    modules: Shared<ModuleList<S>>,
 }
 
 impl<S: Default + ProvableStore + 'static> BaseCoinApp<S> {
     /// Constructor.
     pub(crate) fn new(store: S) -> Result<Self, S::Error> {
```

Comment:
Using the builder pattern here seems appropriate, given that the `BaseCoinApp` is not "initialized" in RAII terms. We could open a separate issue for it though.